contracts/src/v0.8/keystone/lib/KeystoneFeedDefaultMetadataLib.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+library KeystoneFeedDefaultMetadataLib {
+  /**
+   * Metadata Layout:
+   *
+   * +-------------------------------+--------------------+---------------------+---------------+
+   * | 32 bytes (length prefix)      | 32 bytes           | 10 bytes            | 20 bytes      | 2 bytes        |
+   * | (Not used in function)        | workflow_cid       | workflow_name       | workflow_owner| report_name    |
+   * +-------------------------------+--------------------+---------------------+---------------+----------------+
+   * |                               |                    |                     |               |                |
+   * |          (Offset 0)           |     (Offset 32)    |     (Offset 64)     |  (Offset 74)  |  (Offset 94)   |
+   * +-------------------------------+--------------------+---------------------+---------------+----------------+
+   * @dev used to slice metadata bytes into workflowName, workflowOwner and report name
+   */
+  function _extractMetadataInfo(
+    bytes memory metadata
```

Comment:
Currently this is only used in a path where metadata is calldata. It being calldata means we don't have to do assembly and can use normal indexing.

```
    workflowName = bytes10(metadata[0:10]);
    workflowOwner = address(uint160(uint256(bytes32(metadata[32:52]))));
    reportName = bytes2(metadata[94:96]);
    return (workflowName, workflowOwner, reportName);
```

Also, will this be used in other places? If not, it should be in the contract that uses it, not it's own lib

Reply:
I have introduced this library as I am not sure if `metadata` in `onReport` function only consists of the report's workflowName, workflowOwner, reportName only. If this is named metadata then it can encode different information for different use-cases. This is why I have used the keyword "Default" in library's name considering this is the default and most minimal metadata which can be sent in a keystone's report. So whichever contract opts for default metadata can use this can directly .

Reply:
> Also, will this be used in other places? If not, it should be in the contract that uses it, not it's own lib


`KeystoneFeedsConsumer` is an example , `PriceRegistry` will use it and other contracts implementing IReceiver might use it

Reply:
+1 for inlining and using calldata indexers. We can abstract if/when we have more use cases. KeystoneFeedsConsumer is not used in prod and could be dropped entirely if that makes things easier.

Reply:
Signature for metadata:
```
{                      //   (total of 105 bytes)
version,                  // 1 byte
workflow_execution_id,    // 32 bytes, globally unique
timestamp,                // 4 bytes (uint32, big-endian) 
don_id,                   // 4 bytes
don_config_version,	      // 4 bytes (uint32, big-endian)
workflow_cid,             // 32 bytes, hash of the workflow spec
workflow_name,            // 10 bytes, chosen by workflow owner
workflow_owner,           // 20 bytes
report_id,                // 2 bytes
}
```

Reply:
If this is supposed to be a fix set of fields then why aren't we keeping this as a call data struct ?